### PR TITLE
[FEATURE] introduce the snippets into the argument_parser module.

### PR DESCRIPTION
**Pull request by joshuak94**
Also adds type deduction guide in the validators.hpp regex_validator.

### CODE REVIEW

**Comment by marehr** [comment]
can you rebase on top of master again?

File: test/snippet/argument_parser/validators_1.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
+//! [usage]
+#include <seqan3/argument_parser/all.hpp>
+int main(int argc, const char ** argv)
```
Comment (smehringer) [inline test/snippet/argument_parser/validators_1.cpp]: add a new line before main function

**Comment by joshuak94** [comment]
Rebased on current master.

**Comment by marehr** [comment]
@smehringer We could, but there was always the problem that not every output is deterministic.

For example floats and doubles can have different outputs depending on the machine.

This feature request is out of scope of this pr. Can you create an issue?

**Pull request merged**